numba/core/types/abstract.py

```diff
@@ -22,12 +22,14 @@ def _autoincr():
     assert n < 2 ** 32, "Limited to 4 billion types"
     return n
 
-_typecache: ptDict[weakref.ref, weakref.ref] = {}
+_typecache: pt.Dict[weakref.ref, weakref.ref] = {}
 
 def _on_type_disposal(wr, _pop=_typecache.pop):
     _pop(wr, None)
 
 
+_TypeClassType = pt.TypeVar("_TypeClassType")
+
```

Comment:
based on the discussion on August 25 on the Numba-dev gitter channel, it would be useful to distinguish between Numba type classes and instances. I would add in this section (or above)

```
NumbaTypeClass = pt.Type["Type"]
NumbaTypeInst = pt.Union["Type"]
```

Reply:
I like that idea.  Done.  I also updated the section about type annotations in the contributing doc.